agent/api/task_unix.go

```diff
@@ -25,6 +38,70 @@ func getCanonicalPath(path string) string { return path }
 
 // CgroupEnabled checks whether cgroups need be enabled at the task level
 func (task *Task) CgroupEnabled() bool {
```

Comment:
Not all tasks need cgroups even when the feature is enabled at the agent level

Reply:
thanks for clarifying that.

Reply:
@vsiddharth We'll need to have cgroups for every task when the feature is enabled.  We'll also need to disable the feature if there are any running tasks that were created with an old version of the agent.

Because cgroup subsystem settings are inherited (including things like memory and relative weights like CPU shares), we need to bring some of the values up from the container-level cgroups to the parent task-level cgroup or ensure that the parent task-level cgroup has no limit.

This is somewhat easy to reason about with memory, as a limit that is lower than the limit on a child cgroup means that the child will not be able to even reach its limit.  However, the naive approach of summing the child limits to make the parent limit is inflexible and does not allow for overcommit.  Instead, we'll likely want to ensure two things:

1. When no memory limit is specified at the task level, ensure that the parent cgroup has no limit
2. When a memory limit is specified at the task level, ensure that it is greater or equal to the specified hard limit of each individual container (when no hard limit is specified, the task-level limit can be the only limit in place)

For CPU shares, the necessity to bring up container-level CPU shares is a bit less obvious.  There are two things that are in play here:

1. ECS uses the `cpu` field in the task definition as a placement/scheduling constraint, which requires that each container's CPU share is relative to the other containers placed by ECS
2. When a parent cgroup is present, the child cgroup's CPU share is relative to the other children of the same parent and the aggregate of those children is inside the parent's CPU share which is relative to other cgroups at its level.  Another way to think about this is that cgroups' CPU shares are only relative to other cgroups that share the same direct parent

Thus, to preserve the placement property that ECS relies upon, we need to ensure that the aggregate of the containers' CPU share is relative at the task level to other tasks.  We can do this by summing the CPU share of all containers in a task and assigning that CPU share value to the task-level cgroup.